### PR TITLE
fix(issue#25182): eventhubs processor client start span independently

### PR DESCRIPTION
**Pull request by ZejiaJiang**
processor can start span if events come from upstream with opentelemetry is not enable

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by ZejiaJiang** [comment]
fix https://github.com/Azure/azure-sdk-for-java/issues/25182

**Comment by ZejiaJiang** [comment]
@lmolkova I fixed this https://github.com/Azure/azure-sdk-for-java/issues/25182 issue and I found the same logic code in service bus. Should I fix that code as well in this PR?

**Comment by lmolkova** [comment]
> @lmolkova I fixed this #25182 issue and I found the same logic code in service bus. Should I fix that code as well in this PR?

this looks great! Can you please also add a test?

And yes, if you can do servicebus in this PR (or a different one, as you prefer), it would be great.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java [hunk 1]
```diff
@@ -366,11 +367,12 @@ private void cleanup(PartitionOwnership claimedOwnership, PartitionPump partitio
...
     private Context startProcessTracingSpan(EventData eventData, String eventHubName, String fullyQualifiedNamespace) {
         Object diagnosticId = eventData.getProperties().get(DIAGNOSTIC_ID_KEY);
-        if (diagnosticId == null || !tracerProvider.isEnabled()) {
+        if (tracerProvider == null || !tracerProvider.isEnabled()) {
             return Context.NONE;
         }
 
-        Context spanContext = tracerProvider.extractContext(diagnosticId.toString(), Context.NONE)
+        Context spanContext = Objects.isNull(diagnosticId) ? Context.NONE : tracerProvider.extractContext(diagnosticId.toString(), Context.NONE);
+        spanContext = spanContext
```
Comment (samvaity) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java:375]: Will this `diagnosticId.toString()` result in an NPE when diagnosticId is null?

Comment (ZejiaJiang) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java:375]: > Will this `diagnosticId.toString()` result in an NPE when diagnosticId is null?

the former code check diagnosticId before call the toStirng() method. The new code use Objects.IsNull() to check diagnosticId. BTW this PR is closed. And here is the new opened PR https://github.com/Azure/azure-sdk-for-java/pull/26180

**Pull request closed without merging**